### PR TITLE
#162948911 Enable Social login via Google, Facebook and Twitter

### PR DESCRIPTION
**Pull request by Deekerubo**
**What does this PR do?**
This PR enables social Login using Facebook, Twitter and Google accounts. 

**Description of Task to be completed?**
A user cannot Login with a pre-existing social media account. This PR can:
-Test Facebook authentication
-Test Twitter authentication
-Test Google authentication
Users can be able to get their personal information from these social sites. The information enables a user to be authenticated to other social sites and navigate it.

**How should this be manually tested?**
Clone this [repository](https://github.com/andela/ah-alpha.git) then navigate to `ft-social-auth-162948911`
Set the environment, update the requirements.txt file and configure the database.
Get token from:
[Facebook Developer](https://developers.facebook.com/apps)
[Twitter Developer](https://developer.twitter.com/en/apps)
[Google Developer](https://developers.google.com/oauthplayground)

In Your Postman pass the provider and access_token under this url `http://127.0.0.1:8000/api/v1/social_auth/`
Get the access_token from:
[Google Developer](https://developers.google.com/oauthplayground)
```
{
	"provider": "google-oauth2",
     "access_token": ""
}
```
[Facebook Developer](https://developers.facebook.com/apps)
```
{
	"provider": "facebook",
     "access_token": ""
}
```
[Twitter Developer](https://developer.twitter.com/en/apps)
```
{
	"provider": "twitter",
     "access_token": ""
}
```
The output should be a dictionary returning the name, email and access_token from the provider as below;
```
{
    "user": {
        "email": "alphaandela@gmail.com",
        "username": "alphaandela",
        "token": "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpZCI6bnVsbCwiaWF0IjoxNTQ4MzI3MzA1fQ.zEsh9MvDM5uFzjQoklOu-SIGaA6vODN7k36n1Bz6RdY"
    
}
```

**What are the relevant pivotal tracker stories?**
[#162948911](https://www.pivotaltracker.com/story/show/162948911)

### CODE REVIEW

File: authors/apps/authentication/backends.py [hunk 1]
```diff
@@ -1,11 +1,13 @@
...
+import os
 from rest_framework_jwt.settings import api_settings
 import datetime
+from datetime import datetime, timedelta
 import jwt
 from django.conf import settings
 from rest_framework import exceptions
 from rest_framework.authentication import TokenAuthentication
 secret_key = settings.SECRET_KEY
+SECRET_KEY = settings.SECRET_KEY
```
Comment (boltonotieno) [inline authors/apps/authentication/backends.py]: There are two secret keys with the same value. You can use one

File: authors/apps/authentication/tests/test_social_auth.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import os
+from rest_framework.test import APIClient, APITestCase
+from django.urls import reverse
+from django.test import TestCase
+from rest_framework import status
+# from rest_framework import status
```
Comment (boltonotieno) [inline authors/apps/authentication/tests/test_social_auth.py]: Remove the commented line 6

**Comment by danuluma** [comment]
Please fix the merge conflicts

**Review by danuluma — APPROVED** [review]

**Review by boltonotieno — CHANGES_REQUESTED** [review]
You have too many blank lines at the end of the test file. Kindly remove them

File: authors/apps/authentication/backends.py (hunk 1 above)
Comment (Deekerubo) [reply authors/apps/authentication/backends.py]: Thank you. Done!

**Review by boltonotieno — APPROVED** [review]

**Pull request merged**